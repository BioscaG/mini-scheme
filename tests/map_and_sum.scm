(define (map func llista)
  (if (null? llista)
      '()
      (cons (func (car llista)) (map func (cdr llista)))))

(define (dobla x) (* x 2))

(define (sumar-llista llista)
  (if (null? llista)
      0
      (+ (car llista) (sumar-llista (cdr llista)))))

(define (main)
  (let ((llista '(1 2 3 4 5)))
    (display "Llista original: ")
    (display llista)
    (newline)
    (display "Llista doblada: ")
    (display (map dobla llista))
    (newline)
    (display "Suma de la llista doblada: ")
    (display (sumar-llista (map dobla llista)))
    (newline)))