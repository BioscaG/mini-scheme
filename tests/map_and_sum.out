Llista original: (1 2 3 4 5)
Llista doblada: (2 4 6 8 10)
Suma de la llista doblada: 30
